defs/params.go

```diff
@@ -6,66 +6,86 @@ import (
 	ListenerLineBufferSize = InputLogMaxMessageBytes * 4
 
 	// IntermediateBufferMaxNumLogs defines the maximum numbers of log records to buffer at input before flushing through go channels
+	//
 	// The value affects size of buffers passing down channels
+	//
 	// Larger number puts pressure on GC and makes context switching worse,
+	//
 	// e.g. 50000 causes +2.5 sec than 1000 for 10M small logs or 2.3GB, while transform takes less 7 sec
+	//
 	// Related to GO scheduler and internals, re-evaluate the value for go upgrade
 	IntermediateBufferMaxNumLogs = 500
 
 	// IntermediateBufferMaxTotalBytes defines the how many bytes can be allowed in the buffer of input log records before forced flushing
 	IntermediateBufferMaxTotalBytes = 4 * 1024 * 1024
 
 	// IntermediateBufferedChannelSize defines the size of internal buffered channels meant to contain temporary data
+	//
 	// 0 = unbuffered channels
+	//
 	// Higher value improves parallelization when ByKeySetParallelOrchestrator is used
 	IntermediateBufferedChannelSize = 1
 
 	// IntermediateChannelTimeout defines the timeout of intermediate channel reads and writes.
+	//
 	// There is no recovery without data loss and it should be treated as a bug if such timeout happens at runtime
 	IntermediateChannelTimeout = 20 * time.Second
 
 	// IntermediateFlushInterval defines how often for intermediate workers to flush their own states
+	//
 	// For example, to flush buffer streams into output chunks, or to update internal timer
 	IntermediateFlushInterval = 1 * time.Second
 
 	// ParallelizationBufferMaxNumLogs defines the numbers of logs to process before switching to next parallel worker
+	//
 	// Lower value improves parallelization but reduces the size of output chunks
+	//
 	// e.g. 25000 logs * 250 bytes avg / 20 compression ratio ~= 200-300KB chunk(s)
 	ParallelizationBufferMaxNumLogs = 25000
 
 	// ParallelizationBufferMaxTotalBytes defines the numbers of raw input bytes to process before switching to next parallel worker
+	//
 	// The value can be slightly higher than OutputChunkMaxDataBytes
 	ParallelizationBufferMaxTotalBytes = OutputChunkMaxDataBytes + OutputChunkMaxDataBytes/4
 
 	// OutputChunkMaxDataBytes defines the max uncompressed data size of a LogChunk, not including necessary headers
+	//
 	// The value must be lower than the maximum buffer length acceptable by upstream
 	OutputChunkMaxDataBytes = 8 * 1024 * 1024
 
 	// BufferMaxNumChunksInQueue is the max numbers of of loaded and unloaded chunks to be held in a queue,
 	// equal to the max numbers of queued files on disk, because at least all the filepaths need to be held in channel
+	//
 	// New logs are dropped when the limit is reached
-	// e.g. 200000 * 300K chunk files ~= 56GB compressed or 1.1TB uncompressed
-	BufferMaxNumChunksInQueue = 200000
+	//
+	// e.g. 500000 * 300K chunk files ~= 143GB compressed or 3TB uncompressed
+	BufferMaxNumChunksInQueue = 500000
```

Comment:
Should we make this configurable/coming from environment with default value?

Reply:
Yes it can of course be made configurable. But there are dozens of parameters like that and some inter-dependencies that will need to be checked: issue https://github.com/relex/slog-agent/issues/7